testplan/runnable/base.py

```diff
@@ -601,50 +602,60 @@ def _create_result(self):
             else:
                 test_results[uid] = resource_result
 
-            report = test_results[uid].report
+            run, report = test_results[uid].run, test_results[uid].report
+
             if report.part and self.cfg.merge_scheduled_parts:
-                # Save the report temporarily and later will merge it
-                test_rep_lookup.setdefault(report.uid, []).append(
-                    (test_results[uid].run, report)
-                )
                 try:
                     test_report.get_by_uid(report.uid)
-                    continue
                 except KeyError:
+                    # Save the report temporarily and later will merge it
```

Comment:
This Try/expect/else is definetly a if else written in an uggly way ()handling conditional with exception is not performant, as creating teh stack trace and all the things for an exception is huge

if report.id not in test_report:
...
else:
...

Reply:
if report.id in test_report.entry_uids:
  ...
else:
  ...

Use if-esle instead of try-except